src/databricks/labs/lsql/dashboards.py

```diff
@@ -204,16 +204,33 @@ def create_dashboard(self, dashboard_folder: Path) -> Dashboard:
         lakeview_dashboard = Dashboard(datasets=datasets, pages=[page])
         return lakeview_dashboard
 
-    @staticmethod
-    def _get_datasets(dashboard_folder: Path) -> list[Dataset]:
+    def _get_datasets(self, dashboard_folder: Path, *, database: str = "") -> list[Dataset]:
         datasets = []
         for query_path in sorted(dashboard_folder.glob("*.sql")):
-            with query_path.open("r") as query_file:
-                raw_query = query_file.read()
-            dataset = Dataset(name=query_path.stem, display_name=query_path.stem, query=raw_query)
+            query = self._replace_database_in_query(query_path.read_text(), database)
+            dataset = Dataset(name=query_path.stem, display_name=query_path.stem, query=query)
             datasets.append(dataset)
         return datasets
 
+    @staticmethod
+    def _replace_database_in_query(query: str, database: str) -> str:
+        if not database:
+            return query
+
+        try:
+            syntax_tree = sqlglot.parse_one(query, dialect=sqlglot.dialects.Databricks)
+        except sqlglot.ParseError as e:
+            logger.warning(f"Parsing {query}: {e}")
+            return query
+
+        def transformer(node: sqlglot.Expression) -> sqlglot.Expression:
+            if isinstance(node, sqlglot.exp.Table) and node.args.get("db") is not None:
+                node.args["db"].set("this", database)
+            return node
+
+        query_transformed = syntax_tree.transform(transformer).sql(dialect=sqlglot.dialects.Databricks)
```

Comment:
Note that this is not guarantee the same output as input sql (with just database names replaced):

``` python
> import sqlglot
> query = "SELECT * FROM table JOIN another_table"
> sqlglot.parse_one(query, dialect=sqlglot.dialects.Databricks).sql(dialect=sqlglot.dialects.Databricks) == query
False
```